runtime/native.go

```diff
@@ -337,7 +337,27 @@ func getNativeType(rtype reflect.Type) *Type {
 				d[meth.Name] = newNativeMethod(meth.Name, meth.Func)
 			}
 		}
-		t = &newNativeType(rtype, base, newStringDict(d)).Type
+
+		// Building a field for this type depends on the type itself, but we
+		// can't construct the type with the real `*Dict` yet, because we
+		// couldn't otherwise add the new fields to the `*Dict` (because we
+		// don't have access to a `*Frame` in this function). Instead, we'll
+		// convert `d` into a `*Dict` after we construct all of its fields, and
+		// we'll stick it back on `t` at that point.
+		t = &newNativeType(rtype, base, nil).Type
```

Comment:
The tests pass without manually setting the `dict`, so I left it off.

![bitmoji](https://render.bitstrips.com/v2/cpanel/9422789-125772919_2-s1-v1.png?transparent=1&width=246)